src/server/maintenance/mute.ts

```diff
@@ -53,16 +54,57 @@ export default class GameMute extends System {
     const player = this.storage.playerList.get(playerId);
     const playerToMute = this.storage.playerList.get(playerToMuteId);
 
-    if (player.times.activePlaying < CHAT_MIN_PLAYER_PLAYTIME_TO_VOTEMUTE_MS) {
+    /**
+     * Excluding players with no skin in the game.  
+     * A player must play for at least N minutes, and must have a score in the Nth percentile of all human players.
+     */
+    if (player.times.activePlaying < this.config.chat.votemuteDuration) {
       this.emit(
         RESPONSE_COMMAND_REPLY,
         this.storage.playerMainConnectionList.get(playerId),
         `The vote isn't counted. Only active players can vote, please play more.`
       );
+      return;
+    }
 
+    /**
+     * Building a scores array and compare the player's score to the total. 
+     */
+    let scoreToBeat = -1;
+    let scores = [];
+    this.storage.playerList.forEach((p: Player, _: PlayerId) => {
+      scores.push(p.score.current);
+    })
+    // JS casts to unicode before sorting. 
+    scores.sort((a, b) => {
+      return (a > b) ? 1 : -1
+    })
```

Comment:
There is [`playerRankings`](https://github.com/wight-airmash/ab-server/blob/d614031dcd5624bc4e0d16d75009c1d8c81116a9/src/server/storage.ts#L289-L292)  in shared `storage` with `byBounty` array property containing sorted items by scores in descending order.

Reply:
I added a function to the RankingsStorage interface to retrieve a target score from the existing byBounty array.  The function is defined within playerRankings. Not sure if this is the right pattern but it seems more contained than the previous approach, and it doesn't require creating a whole new array.

Reply:
Thank you! Yes, I prefer to keep the data in storage separate from the business logic, so I'll relocate `scoreAtPercentile` func.